shumai/tensor/register_gradients.ts

```diff
@@ -210,6 +210,18 @@ const impls = {
     const o = sm.scalar(1).sub(grad.out)
     return grad.out.mul(o)
   },
+  clip: (grad: Grad) => {
```

Comment:
Still getting the hang of how gradients are implemented in Shumai, so not certain this is correct. Probably wise to eventually have tests around these gradient functions.